### PR TITLE
Bug 1189078 - Add unregister function to remove event handlers and preference observer for copy/paste

### PR DESCRIPTION
**Pull request by chenpighead**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b4bce537a23ab2666bda5781179674ead93f475f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b4bce537a23ab2666bda5781179674ead93f475f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b4bce537a23ab2666bda5781179674ead93f475f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2f42cd8aa1ac926c06bf220df9ab9b993b0e4d23

File: apps/system/js/app_text_selection_dialog.js [hunk 1]
```diff
@@ -67,6 +61,22 @@
...
   AppTextSelectionDialog.prototype.ELEMENT_PREFIX = 'textselection-dialog-';
 
+  AppTextSelectionDialog.prototype.onObservePrefChanged =
+    function tsd_onObservePrefChanged(value) {
+      if (value) {
+        this.start();
+      } else {
+        this.stop();
+      }
+    }.bind(this);
```
Comment (timdream) [inline apps/system/js/app_text_selection_dialog.js]: I don't think `this` is defined here since the instance is not created yet at all.

Comment (timdream) [inline apps/system/js/app_text_selection_dialog.js]: are you sure the feature is still working with this patch?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8bf93bd90da0ad7b640e2f8fd71fd78522d78908

File: apps/system/js/app_text_selection_dialog.js (hunk 1 above)
Comment (chenpighead) [inline apps/system/js/app_text_selection_dialog.js]: Thanks for the feedbacks. I've written a new version and made sure that the feature is still working on my local device.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8bf93bd90da0ad7b640e2f8fd71fd78522d78908

**Pull request merged**